### PR TITLE
Update to v1.5.0

### PR DESCRIPTION
**Pull request by marcoesters**
orjson -> libgit2

The new rust version does not compile `cargo-bundle-licenses` with the current `libgit` version

# Package Info

Home: https://libgit2.org/
Development: https://github.com/libgit2/libgit2
Documentation: https://libgit2.org/docs/
Diff: https://github.com/libgit2/libgit2/compare/v1.0.1...v1.5.0

### CODE REVIEW

**Review by varlackc — COMMENTED** [review]
The following adjustments are highly recommended:

doc_source_url: https://github.com/libgit2/libgit2/tree/main/docs

license_url: https://github.com/libgit2/libgit2/blob/main/COPYING

File: recipe/bld.bat [hunk 1]
```diff
@@ -9,6 +9,7 @@ cmake ^
 	-DLIB_INSTALL_DIR=%LIBRARY_LIB%     ^
 	-DINCLUDE_INSTALL_DIR=%LIBRARY_INC% ^
 	-DSTATIC_CRT=off                    ^
+        -DUSE_SSH=ON                        ^
 	..
 
 ninja install
```
Comment (skupr-anaconda) [inline recipe/bld.bat]: Can this speed up the building time https://github.com/anaconda-distribution/distro-incubator/blob/50336807b4d23581a442dc4ef7677ba5abfd278a/skel/build.sh#L21-L35 ?
The maintainers use cores too https://github.com/libgit2/libgit2/blob/fbea439d4b6fc91c6b619d01b85ab3b7746e4c19/ci/build.sh#L73

File: recipe/build.sh [hunk 2]
```diff
@@ -8,6 +8,7 @@ cmake .. \
     -DCMAKE_FIND_ROOT_PATH=$PREFIX \
     -DCMAKE_INSTALL_LIBDIR=lib \
     -DREGEX_BACKEND=pcre2 \
-    -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_BUILD_TYPE=Release \
+    -DUSE_SSH=ON
```
Comment (skupr-anaconda) [inline recipe/build.sh:12]: Should we adjust CMake options for our build? See https://github.com/libgit2/libgit2/blob/fbea439d4b6fc91c6b619d01b85ab3b7746e4c19/.github/workflows/main.yml#L158

Comment (marcoesters) [reply recipe/build.sh:12]: Not sure. I would say if it ain't broke, don't fix it?

**Review by skupr-anaconda — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**